### ISSUE TITLE
perl warning Odd number of elements in anonymous hash

### ISSUE THREAD

**Issue by kenshin33** [OCSInventory-NG/UnixAgent#310]
### General informations
Operating system :  Gentoo Linux
Perl version : 5.30.3

### OCS Inventory informations
Unix agent version : 2.8

### Problem's description
when the number of memebers in a given group is even I nooticed this warning when I ran the agent 
Odd number of elements in anonymous hash at /usr/local/lib64/perl5/5.30.3/Ocsinventory/Agent/Backend/OS/Generic/Users.pm line 109

this is the source of the warning https://github.com/OCSInventory-NG/UnixAgent/blob/master/lib/Ocsinventory/Agent/Backend/OS/Generic/Users.pm#L112

It is an incorrect usage of => in that context, as the array will be flattened with will create other key/values in the hash (most probably not what is expected)
for example those are the keys for hash corresponding to the daemon group on my system:

    daemon:
              ID MEMBER NAME bin
    cat /etc/group | grep '^daemon'
    daemon:x:2:root,bin,daemon

**Comment by daniejstriata** [comment]
I get it too:
```
# ocsinventory-agent -f
Odd number of elements in anonymous hash at /usr/share/perl5/vendor_perl/Ocsinventory/Agent/Backend/OS/Generic/Users.pm line 109.
Odd number of elements in anonymous hash at /usr/share/perl5/vendor_perl/Ocsinventory/Agent/Backend/OS/Generic/Users.pm line 109.
```

**Comment by fbomj** [comment]
Fix in PR #315

**Comment by githubdanieldiehl** [comment]
**Solução/solution:**

sudo nano /usr/share/perl5/Ocsinventory/Agent/Backend/OS/Generic/Users.pm

**ADDED**:
Prints...
If (@members)
\@members


**Change**:

sub _getLocalGroups {
 
     open(my $fh, '<:encoding(UTF-8)', "/etc/group") or warn;
     my @groupinfo=<$fh>;
     close($fh);
 
     my @groups;
 
     foreach my $line (@groupinfo){
         next if $line =~ /^#/;
         chomp $line;
         my ($name, undef, $gid, $members) = split(/:/, $line);
 
         next unless $members;
         my @members = split(/,/, $members);
         use Data::Dumper;
 
         print "Group ID: $gid\n";
         print "Group Name: $name\n";
         print "Members: ", Dumper(\@members), "\n";
 
     if (@members) {
         push @groups, {
             ID     => $gid,
             NAME   => $name,
             MEMBER => \@members,
         };
      }
      }
 
     return @groups;
 
}
